x509/zintermediate/zintermediate.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * ZCrypto Copyright 2017 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package main
+
+import (
+	"bufio"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+
+	"github.com/op/go-logging"
+)
+
+var log = logging.MustGetLogger("")
+
+var inputFormatArg string
+
+type inputFormatType int
+
+const (
+	inputFormatBase64 inputFormatType = iota
+	inputFormatPEM
+	inputFormatJSON
+)
+
+var inputFormat inputFormatType
+var inputFileName, rootFileName string
+
+func init() {
+	flag.StringVar(&inputFormatArg, "format", "base64", "One of {base64, pem, json}")
+	flag.StringVar(&rootFileName, "roots", "roots.pem", "Path to root store")
+	flag.Parse()
+
+	if flag.NArg() < 1 {
+		log.Fatalf("missing filename")
+	}
+	inputFileName = flag.Arg(0)
+
+	inputFormatArg = strings.ToLower(inputFormatArg)
+	switch inputFormatArg {
+	case "base64":
+		inputFormat = inputFormatBase64
+	case "pem":
+		inputFormat = inputFormatPEM
+	case "json":
```

Comment:
Because you get certs from Censys. It would just read `raw`, if I got around to it.